pkg/webhook/validator.go

```diff
@@ -1012,3 +1039,90 @@ func refOrFieldError(image, field string, index int) (name.Reference, *apis.Fiel
 	}
 	return ref, nil
 }
+
+// configFileResult is used to communicate results from gofuncs that fetch
+// ConfigFiles for a given image.
+// Because this can be recursive (say, multi-arch image), returns a map where
+// key is the architecture of the container image.
+type configFileResult struct {
+	ret  map[string]*v1.ConfigFile
+	errs []error
+}
+
+// getConfigs will fetch ConfigFile(s) for a given image. In case the image
+// is an index, we'll fetch the arch images recursively.
+func getConfigs(ctx context.Context, ref name.Reference, options ...remote.Option) (map[string]*v1.ConfigFile, []error) {
+	descriptor, err := remote.Get(ref, options...)
+	if err != nil {
+		return nil, []error{fmt.Errorf("failed to get ref %s : %w", ref.String(), err)}
+	}
+	switch descriptor.MediaType {
+	case types.OCIImageIndex, types.DockerManifestList:
+		ii, err := descriptor.ImageIndex()
+		if err != nil {
+			return nil, []error{fmt.Errorf("getting ImageIndex for %s : %w", ref.String(), err)}
+		}
+		im, err := ii.IndexManifest()
+		if err != nil {
+			return nil, []error{fmt.Errorf("getting IndexManifest for %s : %w", ref.String(), err)}
+		}
+		wg := new(sync.WaitGroup)
+
+		results := make(chan configFileResult, len(im.Manifests))
+		for _, manifest := range im.Manifests {
+			manifest := manifest
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+
+				newRefString := fmt.Sprintf("%s/%s@%s", ref.Context().RegistryStr(), ref.Context().RepositoryStr(), manifest.Digest)
```

Comment:
I think `Context()` has a `Digest(string)` method

Reply:
What I mean is that `ref.Context()` is a `name.Repository` which has this method: https://github.com/google/go-containerregistry/blob/bdc946f4d482443524054e63fc56c11811858cae/pkg/name/repository.go#L114

So I think this could be: `ref.Context().Digest(manifest.Digest)`